### DISCUSSION TITLE
Understanding Project Requirements and Queries

### DISCUSSION THREAD

**Discussion by yusuf-khaan** [camicroscope/caMicroscope#941] [Q&A]
Greeting! 

Hey! @birm , I have applied for Gsoc candidacy to include multi channel support for spatial representation for more than three channels. I was going through all the repositories and excited to include such feature. I was hoping you could tell what are the repositories that needs to cloned and actions have to be performed, Your Guidance would be critical for me to Contribute and get it done in soonest time frame. Please Provide feedback for it 😊

The Repositories that i have visited untill now are 

[caMicroscope Distro](https://github.com/camicroscope/distro)

[caMicroscope Viewer](https://github.com/camicroscope/caMicroscope)

[caMicroscope Caracal](https://github.com/camicroscope/Caracal)

Do i need any more than this??Your Guidance would be utterly Critical and Informative

**Comment by yusuf-khaan** [comment]
hey @birm @gbengaoluwadahunsi 
![Screenshot 2024-04-04 121841](https://github.com/camicroscope/caMicroscope/assets/142320884/68222d07-d539-4c2a-9e7e-0a5c8b261c82)

Can you tell it shows this error after composing develop.yml file in docker

**Reply by gbengaoluwadahunsi** [reply]
Hello @yusuf-khaan ,
No, mine is  NOT showing that error when  I ran it on my local machine.

**Reply by yusuf-khaan** [reply]
Any way of resolving it

**Reply by birm** [reply]
@yusuf-khaan does it persist if you try to build again? can you see ubuntu's security page at 91.189.91.81 (or just ping it from wherever you're building)

**Reply by gbengaoluwadahunsi** [reply]
@birm

Please, I want to ask as regards some of us that have open PRs, will it count as a contribution?

I quite understand you are busy too, thanks.

**Reply by yusuf-khaan** [reply]
@birm I have one really important Question please response whenever you get time

**High Dimensional Imaging Support**
_"Sometimes, the spatial data represented is not the direct RGB values from a scanner. In this case, we encode three values as RGB."_

Do you meant to say images such as 
**High Dimensional Images** - such as hyperspectral Images -- _where each pixel contains information across spectral bands to provide more detailed spectral signatures compared to RGB imaging_
**Multi-Channel Image Support** which is mentioned in the description - which may include CMYK, LAB(CIELAB) color spaces
or do you mean **fluorescence imaging channels**

Sorry for asking too much question but i need to wrap my head around to start working when i am analyzing the architecture and working on it

**Reply by yusuf-khaan** [reply]
No it does not persist now maybe it was some handling error

Thanks

**Comment by gbengaoluwadahunsi** [comment]
I think this is similar to what you want to achieve
https://github.com/camicroscope/GSOC/discussions/45

for how to get the caMicroscope up and running this is what you should do

download  and install docker on your local machine and ensure its running

Use               docker-compose -f develop.yml build loader --no-cache (to rebuild the services)
Start with     docker-compose -f caMicroscope.yml up


Once everything is up, go to http://localhost:4010/ to see the landing page.

**Reply by yusuf-khaan** [reply]
Thanks for your response! Really appreciated

**Reply by gbengaoluwadahunsi** [reply]
You are welcome @yusuf-khaan

**Comment by yusuf-khaan** [comment]
Greetings @birm @gbengaoluwadahunsi 

Do you have any .envi / .tiff or any hyper spectral images sample with their data files to test my model to include High Dimensional Image Support , or any site where i can available such datasets

**Comment by gbengaoluwadahunsi** [comment]
I will admit it's only @birm that can give you an answer on this.

Thanks.
